Blog multidioma con Polyglot

UrkoLekuona opened the pull request:
Esta es la versión actual del blog multidioma usando Polyglot.

He añadido la versión traducida del post de "buenas prácticas para escribir un dockerfile" para que se pueda ver cómo funciona.

Falta por hacer:

- [x] Añadir `lang` al _frontmatter_ de todos los posts (no es necesario pero lo haría por consistencia).
- [x] Traducir las páginas comunes del blog (como [`cookies-message.html`](https://github.com/wearearima/wearearima.github.io/blob/master/_includes/cookies-message.html), [`cookies.markdown`](https://github.com/wearearima/wearearima.github.io/blob/master/cookies.markdown), ...) (no sé si de esto se encarga Janice o yo).

Faltaría también traducir el resto de posts, pero no me parece que sea una tarea de este PR, sino de otro(s) aparte.

Me gustaría que alguien revisase la parte de CSS/HTML del código, porque no soy ningún experto y seguro que se puede mejorar.

Otras opiniones también son bienvenidas :D.

closes #26

UrkoLekuona committed:
Started adding polyglot support
Polyglot installation
Reorganized posts file structure

UrkoLekuona committed:
Added language selector to toolbar and home

UrkoLekuona committed:
Frontmatter lang to dockerfile

UrkoLekuona committed:
Comment about sourcemap never

aberasarte commented on _sass/partials/_toolbar.scss:
```diff
@@ -28,4 +28,19 @@
+    /* Language selector */
+    .language-selector {
+        color: black;
+        font-size: 18px;
```
Yo lo bajaría a 12px.

UrkoLekuona replied:
Cambiado (aunque me parece un poco pequeño).

aberasarte commented:
Yo lo veo bien a falta de tratar lo del permalink en inglés en otro PR.

@uurien échale un vistazo al css/html por si te canta algo.
@itelleria, ¿te encargas de la traducción de las páginas comunes?

Good job @UrkoLekuona

uurien commented on _sass/partials/_toolbar.scss:
```diff
@@ -28,4 +28,19 @@
+        float: right;
+        margin: 1% 0 0 0;
+        .selected {
+          font-family: gothambold,Arial,Helvetica,sans-serif;
```
Esto mismo lo tenemos en esta variable en __common.scss_ `$bold-font-family`

UrkoLekuona replied:
No me había fijado, lo cambio.

uurien commented on _sass/partials/_toolbar.scss:
```diff
@@ -28,4 +28,19 @@
         }
     }
+    /* Language selector */
+    .language-selector {
```
Haría también los siguientes cambios:
```
/* al principio */
$toolbar-height: 40px;
/* en .toolbar, cambiar el height */
height: $toolbar-height;
/* Y en .language-selector, quitar el margin y poner esto */
height: $toolbar-height;
line-height: $toolbar-height;
```
Con esto se centrará bien (habría que probar navegadores)

UrkoLekuona replied:
Es un buen cambio. Yo estaba intentado centrarlo sin tocar nada de otros elementos, pero viendo así mucho mejor.

Lo he probado en Chrome, Mozilla y Safari y para los 3 parece funcionar bien (actualizados todos, versiones anteriores ni idea). De todas formas, [`line-height` lo soportan todos los navegadores desde hace siglos](https://www.w3schools.com/cssref/pr_dim_line-height.asp), ¿no?

UrkoLekuona committed:
Css changes
Toolbar height to center lang selector
Font family as variable
Selector font size

UrkoLekuona committed:
SEO recipes

UrkoLekuona committed:
Lang to all frontmatters

UrkoLekuona commented:
Por mi parte ya está finalizado, falta traducir las páginas comunes que se está encargando @itelleria

UrkoLekuona committed:
Translated common pages
